frontend/src/main/scala/bloop/engine/ExecutionContext.scala

```diff
@@ -16,6 +25,42 @@ object ExecutionContext {
     java.util.concurrent.Executors.newFixedThreadPool(4)
   }
 
-  implicit val scheduler: Scheduler = Scheduler(executor)
-  implicit val ioScheduler: Scheduler = Scheduler.io("bloop-io")
+  implicit lazy val scheduler: Scheduler = Scheduler(executor)
+
+  val ioReporter = UncaughtExceptionReporter.LogExceptionsToStandardErr
+  lazy val ioExecutor: ThreadPoolExecutor = {
+    val threadFactory = monixThreadFactoryBuilder("bloop-io", ioReporter, daemonic = true)
+    new ThreadPoolExecutor(
+      0,
+      Int.MaxValue,
+      60,
+      TimeUnit.SECONDS,
+      new SynchronousQueue[Runnable](false),
+      threadFactory
+    )
+  }
+
+  implicit lazy val ioScheduler: Scheduler =
+    ExecutorScheduler(ioExecutor, ioReporter, ExecutionModel.Default)
+
+  // Inlined from `monix.execution.schedulers.ThreadFactoryBuilder`
```

Comment:
Because the executor needs to be exposed so that https://github.com/scalacenter/bloop/pull/389/files/3b07914214c27039658685c68355de169f083d3c#diff-f9ab286c6a27ce9fefdf1f0373d83486R67 works.